sound/soc/sof/intel/hda.c

```diff
@@ -597,6 +597,8 @@ int hda_dsp_probe(struct snd_sof_dev *sdev)
 	/* set default mailbox offset for FW ready message */
 	sdev->dsp_box.offset = HDA_DSP_MBOX_UPLINK_OFFSET;
 
+	snd_hda_set_power_save(sof_to_hbus(sdev), SND_SOF_SUSPEND_DELAY_MS);
+
```

Comment:
the commit message is far from clear to me, so I don't get the problem statement;

````
Even though SOF sets its own suspend delay, the possible HDA codec of
the driver ignores it.
````

SOF sets is own pm_runtime autosuspend delay?
what does "the possible HDA codec of the driver ignores it." mean? What is ignored by what? Which driver?

````
 So even as SOF sets the suspend delay to 2s, hda
codec will suspend immediately or depending on the value of
CONFIG_SND_HDA_POWER_SAVE_DEFAULT. 
````
Why? Is this because there is no parent-child relationship between the SOF PCI device and the codec device? What causes this behavior? You are describing the result, not the reason for the immediate suspend.

````
This will lead in some cases to jack
pins being detected incorrectly. 
````

Why? what does pin/jack detection have to do with suspend?

````
So set the suspend delay time in HDA
codec to same value as SOF tries to set itself.
````
why does this solve anything? The autosuspend delay on the SOF side is not a strict value, it's subject to kernel scheduling, so I don't see how you avoid a race condition with transitions on the codec side. If you can't control the scheduling, you have to use a significantly larger value on one side so that even in the worst possible case nothing bad happens.